packages/core/src/utils/SIWXUtil.ts

```diff
@@ -1,4 +1,256 @@
 import type { CaipNetworkId } from '@reown/appkit-common'
+import { OptionsController } from '../controllers/OptionsController.js'
+import { CoreHelperUtil } from './CoreHelperUtil.js'
+import { ChainController } from '../controllers/ChainController.js'
+import { ConnectionController } from '../controllers/ConnectionController.js'
+import { ModalController } from '../controllers/ModalController.js'
+import { SnackController } from '../controllers/SnackController.js'
+import { RouterController } from '../controllers/RouterController.js'
+import UniversalProvider from '@walletconnect/universal-provider'
+import { EventsController } from '../controllers/EventsController.js'
+import { AccountController } from '../controllers/AccountController.js'
+import { W3mFrameRpcConstants } from '@reown/appkit-wallet'
+import { StorageUtil } from './StorageUtil.js'
+
+export const SIWXUtil = {
+  getSIWX() {
+    return OptionsController.state.siwx
+  },
+  async initializeIfEnabled() {
+    const siwx = OptionsController.state.siwx
+    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
+    const network = ChainController.getActiveCaipNetwork()
+
+    if (!(siwx && address && network)) {
+      return
+    }
```

Comment:
should we log anything here?

Reply:
I don't think so, this is just ignoring siwx if it is not enabled